plugins/destination/mssql/client/write.go

```diff
@@ -0,0 +1,118 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/plugins/destination"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+)
+
+func (c *Client) Write(ctx context.Context, tables schema.Tables, resources <-chan *destination.ClientResource) error {
+	query := upsert
+	if c.spec.WriteMode == specs.WriteModeAppend {
+		query = insert
+	}
+
+	batchSize := c.spec.BatchSize
+
+	batch := make([]*stmt, 0, batchSize)
+	for r := range resources {
+		table := tables.Get(r.TableName)
+		if table == nil {
+			panic(fmt.Errorf("table %s not found", r.TableName))
+		}
+
+		batch = append(batch, &stmt{query: query(table), params: r.Data})
+		if len(batch) == batchSize {
+			if err := c.send(ctx, batch); err != nil {
+				return err
+			}
+
+			// just clip, capacity is left intact (opposed to batch[:0:0])
+			batch = batch[:0]
+		}
+	}
+
+	return c.send(ctx, batch)
+}
+
+func insert(table *schema.Table) string {
+	var sb strings.Builder
+
+	sb.WriteString("insert into ")
+	sb.WriteString(sanitizeIdentifier(table.Name))
+	sb.WriteString("\n(\n\t")
+
+	columns := tableColumnsSanitized(table)
+	sb.WriteString(strings.Join(columns, ",\n\t"))
```

Comment:
why `\n\t` is need for sql commands?

Reply:
nott required, but allows to read queries and debug them easier